### PR TITLE
Add the regcode, product and instance-data cli options

### PR DESCRIPTION
**Pull request by djoreilly**
This commit just adds these flags to the cli, and checks for
invalid combinations like the Ruby version.

### CODE REVIEW

File: connect/config.go [hunk 1]
```diff
@@ -18,12 +18,15 @@ const (
...
-	Namespace string
-	FsRoot    string
+	Path             string
+	BaseURL          string
+	Language         string
+	Insecure         bool
+	Namespace        string
+	FsRoot           string
+	Token            string
+	Product          string
```
Comment (djoreilly) [inline connect/config.go:28]: I think this should type Product{} not a string.

File: cmd/suseconnect.go [hunk 2]
```diff
@@ -94,6 +115,21 @@ func main() {
 	} else if deregister {
 		err := connect.Deregister()
 		exitOnError(err)
+	} else {
+		if instanceDataFile != "" && connect.URLDefault() {
```
Comment (skazi0) [inline cmd/suseconnect.go:119]: why is this nested and not just another `else if`?

Comment (djoreilly) [reply cmd/suseconnect.go:119]: > why is this nested and not just another `else if`?

I'm trying to follow the flow of the Ruby version:
https://github.com/SUSE/connect/blob/master/lib/suse/connect/cli.rb#L38-L51

Comment (skazi0) [reply cmd/suseconnect.go:119]: hmm... I guess that's fine then.

**Review by skazi0 — APPROVED** [review]
lgtm

File: cmd/suseconnect.go (hunk 2 above)
Comment (djoreilly) [reply cmd/suseconnect.go:119]: I agree it's not nice. Maybe after we have parity of functionality we can come back and improve the whole cli option processing.

**Pull request merged**